spec/features/editing_content_settings/backdate_spec.rb

```diff
@@ -48,29 +41,38 @@ def and_i_enter_a_date_to_backdate_the_content_to
   end
 
   def and_i_click_save
-    @request = stub_publishing_api_put_content(@edition.content_id, {})
+    @set_request = stub_publishing_api_put_content(
+      @edition.content_id,
+      hash_including("first_published_at" => "2019-01-01T00:00:00.000+00:00"),
+    )
     click_on "Save"
   end
 
   def then_i_see_the_content_has_been_backdated
+    expect(@set_request).to have_been_requested
+
     expect(page).to have_content("1 January 2019")
     expect(page).to have_content(I18n.t!("documents.history.entry_types.backdated",
                                          date: "01 January 2019"))
-    expect(@request).to have_been_requested
   end
 
-  def given_there_is_a_document_with_a_second_edition
-    @edition = create(:edition, number: 2)
-  end
-
-  def then_i_cannot_see_a_backdate_section
-    expect(page).not_to have_content(
-      I18n.t!("documents.show.content_settings.backdate.title"),
+  def and_i_click_clear_backdate
+    @clear_request = stub_publishing_api_put_content(
+      @edition.content_id,
+      # hash_excluding doesn't seem to work here
+      ->(body) { JSON.parse(body).exclude?("first_published_at") },
```

Comment:
I have no idea how this works, can you explain it to me at some point? Like I understand its a Lambda but how it integrates with stub_publishing_api_put_content baffles me! 😆🙏

Reply:
Sure, so when a request to this mock is called this lambda is executed by passing in the body string. We then parse this into a hash and check that it is excluding a particular field. Ideally I'd have done `stub_publishing_api_put_content(@edition.content_id, hash_excluding(:first_published))` - however that doesn't work. And Ben looks to have successfully worked out why (🎉) in https://github.com/alphagov/content-publisher/pull/1627#discussion_r364875223